clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/ttl/VeniceRmdTTLFilter.java

```diff
@@ -105,15 +109,22 @@ public void close() {
 
   boolean filterByTTLandMaybeUpdateValue(final INPUT_VALUE value) {
     ByteBuffer rmdPayload = getRmdPayload(value);
-    if (rmdPayload == null || !rmdPayload.hasRemaining()) {
+    boolean hasRmdPayload = rmdPayload != null && rmdPayload.hasRemaining();
+    if (!hasRmdPayload && ttlPolicy == TTLResolutionPolicy.RT_WRITE_ONLY) {
       throw new IllegalStateException(
           "The record doesn't contain required RMD field. Please check if your store has A/A enabled");
     }
     int valueSchemaId = getSchemaId(value);
     int id = getRmdProtocolId(value);
     RmdVersionId rmdVersionId = new RmdVersionId(valueSchemaId, id);
-    GenericRecord rmdRecord =
-        rmdDeserializerCache.computeIfAbsent(rmdVersionId, this::generateRmdDeserializer).deserialize(rmdPayload);
+    final GenericRecord rmdRecord;
+    if (hasRmdPayload) {
+      rmdRecord =
+          rmdDeserializerCache.computeIfAbsent(rmdVersionId, this::generateRmdDeserializer).deserialize(rmdPayload);
+    } else {
+      long logicalTimestamp = getLogicalTimestamp(value);
+      return logicalTimestamp <= filterTimestamp;
```

Comment:
If we use offset, isn't this condition is true most of the time?

Reply:
I made a mistake previously. The timestamp should never use offset. It was a copy-paste error

Just updated